File: Bubblesort/bubblesort.py
def swap(arr,a,b):
    store = arr[a]
    arr[a] = arr[b]
    arr[b] = store

def bubble_sort(arr):
    swapped = True
    length = len(arr) - 1
    while(swapped):
        swapped = False
        for i in range(length):
            if(arr[i]>arr[i+1]):
                swap(arr,i,i+1)
                swapped = True
                print(arr)
        length -= 1

input_string = input("Enter array (comma or space separated): ")

arr = [int(x) for x in input_string.replace(',', ' ').split()]

print("Original array:", arr)

bubble_sort(arr)

print("Sorted array:", arr)

File: Bubblesort/bubblesort.js
//Run in Node.js
function swap(arr,a,b)
{
    let store = arr[a];
    arr[a] = arr[b];
    arr[b] = store;
}

function bubbleSort(arr)
{
    let swapped = true;
    let length = arr.length - 1;
    while(swapped)
    {
        swapped = false;
        for(let i=0; i<length;i++)
        {
            if(arr[i]>arr[i+1])
            {
                swap(arr,i,i+1);
                swapped = true;
                console.log(arr);
            }
        }
        length--;
    }
}

const readline = require('readline').createInterface({
  input: process.stdin,
  output: process.stdout,
});

readline.question('Enter array (comma or space separated): ', (input) => 
{
    let arr = input.split(/[\s,]+/).map(Number);

    console.log('Original array:', arr);

    bubbleSort(arr);

    console.log('Sorted array:', arr);

    readline.close();
});
